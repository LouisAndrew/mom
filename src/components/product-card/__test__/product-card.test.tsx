import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import Card from '..';

describe('Product card component', () => {
    const mockHeadingText: string = 'Heading';
    const mockBodyText: string = 'Body';
    const imgSrc: string =
        'https://res.cloudinary.com/dsvdffre0/image/upload/v1596016031/github-icon.png';
    const mockNavigate = jest.fn(() => {});

    afterEach(cleanup);
    const el: React.ReactElement = (
        <Card
            headingText={mockHeadingText}
            bodyText={mockBodyText}
            src={imgSrc}
            alt="github-icon"
            navigate={mockNavigate}
        />
    );

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(el, div);
    });

    it('should react to user click correctly', () => {
        const { getByRole } = render(el);
        const button: HTMLElement = getByRole('button');

        fireEvent.click(button);
        expect(mockNavigate).toBeCalled();
    });

    it('matches snapshot', () => {
        const tree = renderer.create(el).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
