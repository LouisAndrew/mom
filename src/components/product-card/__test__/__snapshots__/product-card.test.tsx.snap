// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product card component matches snapshot 1`] = `
<div
  className="sc-AxmLO ljJVhi"
>
  <div
    className="sc-AxheI jFHUsp"
    height={
      Array [
        200,
        250,
        300,
      ]
    }
    width={
      Array [
        200,
        250,
        300,
      ]
    }
  >
    <img
      alt="github-icon"
      src="https://res.cloudinary.com/dsvdffre0/image/upload/v1596016031/github-icon.png"
    />
  </div>
  <div
    className="sc-fzozJi fvxFuZ"
    width={
      Array [
        1,
        0.8,
        0.5,
      ]
    }
  >
    <h3
      className="sc-AxiKw ddVZVi"
      fontFamily="heading"
      fontSize={
        Array [
          3,
          3,
          4,
          4,
        ]
      }
    >
      Heading
    </h3>
    <p
      className="sc-AxhUy ffKmGj"
      fontFamily="body"
      fontSize={
        Array [
          1,
          1,
          2,
          2,
        ]
      }
    >
      Body
    </p>
    <button
      className="sc-Axmtr ctITow"
      fontFamily="body"
      fontSize={
        Array [
          1,
          1,
          2,
          2,
        ]
      }
      fontWeight="bold"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [
              Class {
                "_dispatchInstances": null,
                "_dispatchListeners": null,
                "_targetInst": null,
                "altKey": null,
                "bubbles": null,
                "button": null,
                "buttons": null,
                "cancelable": null,
                "clientX": null,
                "clientY": null,
                "ctrlKey": null,
                "currentTarget": [Function],
                "defaultPrevented": null,
                "detail": null,
                "dispatchConfig": null,
                "eventPhase": null,
                "getModifierState": [Function],
                "isDefaultPrevented": [Function],
                "isPropagationStopped": [Function],
                "isTrusted": null,
                "metaKey": null,
                "movementX": [Function],
                "movementY": [Function],
                "nativeEvent": null,
                "pageX": null,
                "pageY": null,
                "relatedTarget": [Function],
                "screenX": null,
                "screenY": null,
                "shiftKey": null,
                "target": null,
                "timeStamp": [Function],
                "type": null,
                "view": null,
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      More info
    </button>
  </div>
</div>
`;
